sdk/core/core/src/az_span.c

```diff
@@ -129,6 +129,78 @@ AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out_span)
+{
+  AZ_PRECONDITION_VALID_SPAN(source, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(target, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
+  uint8_t* source_ptr = az_span_ptr(source);
+  int32_t source_length = az_span_length(source);
+  uint8_t* target_ptr = az_span_ptr(target);
+  int32_t target_length = az_span_length(target);
+
+  for (int i = 0; i < source_length; i++)
```

Comment:
I'd still love to see some comments in here explaining how this code is trying to accomplish its task. Imagine that someone else has to maintain this and there may be a bug or they might need to add a feature to it. Briefly explain the algorithm you're using to solve this problem.

Reply:
No problem at all. Done.